move Akka from 2.4 to 2.5

it's time, 2.5 has been out for a while.

https://github.com/SethTisue/community-builds/commits/akka-2.5 has an initial stab

the test run at https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1829/  shows that unforking akka-http (possible now that they're running modern sbt-bintray) caused some weird OSGi-related OutOfMemoryError (cranking up the heap size didn't help, it must be constructing an infinite string). I should return to it

```
[akka-http:error] Caused by: java.lang.OutOfMemoryError: Java heap space
[akka-http:error] 	at java.util.Arrays.copyOf(Arrays.java:3332)
[akka-http:error] 	at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:124)
[akka-http:error] 	at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:448)
[akka-http:error] 	at java.lang.StringBuilder.append(StringBuilder.java:136)
[akka-http:error] 	at java.lang.StringBuilder.append(StringBuilder.java:76)
[akka-http:error] 	at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:484)
[akka-http:error] 	at java.lang.StringBuilder.append(StringBuilder.java:166)
[akka-http:error] 	at java.lang.StringBuilder.append(StringBuilder.java:76)
[akka-http:error] 	at java.util.Formatter$FormatSpecifier.print(Formatter.java:2769)
[akka-http:error] 	at java.util.Formatter.format(Formatter.java:2508)
[akka-http:error] 	at java.util.Formatter.format(Formatter.java:2455)
[akka-http:error] 	at java.lang.String.format(String.java:2940)
[akka-http:error] 	at aQute.bnd.osgi.Processor.report(Processor.java:1799)
[akka-http:error] 	at aQute.bnd.osgi.Processor.toString(Processor.java:2235)
```

likely culprit is https://github.com/akka/akka-http/pull/1008 — probably this OSGi stuff doesn't like dbuild-mangled version numbers, something like that?

it seems we already dealt with the same thing (probably the same?) in scala/scala, see https://github.com/scala/scala/pull/5581#issuecomment-265001683 and below

possibly relevant: https://github.com/sbt/sbt-osgi/issues/15 , https://github.com/bndtools/bnd/issues/724

note that in the test run, two subprojects succeed (akka-parsing, akka-http-core) before one fails (akka-http), but the successful logs include a great deal of spew like this:

```
[akka-http] [warn] -----------------
[akka-http] [warn] Warnings
[akka-http] [warn] 000: No translation found for macro: range;[===,=+);10.0.0-dbuildxae2a802999ec892c737b3f40cbe55f61d0316ada, in -----------------
[akka-http] [warn] Errors
[akka-http] [warn] 000: Invalid syntax for version: 10.0.0-dbuildxae2a802999ec892c737b3f40cbe55f61d0316ada, for cmd: range, arguments; [range, [===,=+), 10.0.0-dbuildxae2a802999ec892c737b3f40cbe55f61d0316ada]
[akka-http] [warn] bnd: No translation found for macro: range;[==,=+), in -----------------
[akka-http] [warn] Errors
[akka-http] [warn] 000: Invalid syntax for version: 10.0.0-dbuildxae2a802999ec892c737b3f40cbe55f61d0316ada, for cmd: range, arguments; [range, [===,=+), 10.0.0-dbuildxae2a802999ec892c737b3f40cbe55f61d0316ada]
```

we could always fork akka-http and just rip all of the OSGi stuff out, but perhaps there's something we could put in `extra.commands` to bypass the computation that's blowing up... maybe `set every osgiBundleVersion ....`

--- a/configs/community.dbuild
+++ b/configs/community.dbuild
@@ -284,10 +284,9 @@ build += {
     extra.test-tasks: ["compile"]
   }
 
-  // tracking release-2.4 because master is 2.5 and breaking changes are happening
   ${vars.base} {
-    name: "akka"
-    uri:  ${vars.uris.akka-uri}
+    name: "akka-actor"
+    uri:  ${vars.uris.akka-actor-uri}
     extra: ${vars.base.extra} {
       options: ["-Dakka.genjavadoc.enabled=false", "-Dakka.scaladoc.diagrams=false", "-Dakka.build.aggregateSamples=false"]
       projects: ["akka-actor"]
@@ -295,7 +294,6 @@ build += {
     }
   }
 
-  // tracking release-2.4 because master is 2.5 and breaking changes are happening
   // this is separate from "akka" because there is a circular dependency between
   // the akka and ssl-config repos
   ${vars.base} {
@@ -315,16 +313,21 @@ build += {
     }
   }
 
-  // forked (December 2016) to get rid of the usual bintray-sbt stuff that
-  // makes dbuild upset
   ${vars.base} {
     name: "akka-http"
     uri:  ${vars.uris.akka-http-uri}
     extra: ${vars.base.extra} {
-      options: ["-Dakka.genjavadoc.enabled=false", "-Dakka.scaladoc.diagrams=false"]
+      exclude: ["docs"]
+      options: [
+        "-Dakka.genjavadoc.enabled=false", "-Dakka.scaladoc.diagrams=false"
+        "-Dbintray.user=dummy", "-Dbintray.pass=dummy"
+      ]
       // Scaladoc generation failure reported upstream at https://github.com/akka/akka/issues/21543
       commands: ${vars.default-commands} [
         "set sources in doc in Compile in httpCore := List()"
+        "set every bintrayReleaseOnPublish := false"
+        // sbt-osgi doesn't like dbuild-mangled version numbers
+        "set every OsgiKeys.bundleVersion := \"10.0.0\""
       ]
       // "HTTP is sadly very timing sensitive we're working on improving its stability regularly,
       // OK to disable it for now." - Konrad M, October 2016
@@ -851,23 +854,10 @@ build += {
       "lagom13JavaConductRBundleLib", "lagom13ScalaConductRBundleLib"
       // didn't compile (July 2017)
       "akka24ConductRClientLib"
-      // not our Akka version
-      "akka25Common", "akka25ConductRBundleLib", "akka25ConductRClientLib", "akka25TestLib"
       // Play stuff didn't work, maybe try again later?
       "play25ConductRBundleLib", "play25ConductRClientLib", "play25Common"
       "play26ConductRBundleLib", "play26ConductRClientLib", "play26Common"
     ]
-  }
-
-  // can be removed anytime if it ever acts up, as the project is now
-  // obsolete/EOL (reference:
-  // https://github.com/hseeberger/akka-sse/commit/f18a359b489cc6a749709b5ce2d6beaafdd0f32f)
-  // forked (December 2016) to get rid of the usual bintray-sbt stuff that
-  // makes dbuild upset
-  ${vars.base} {
-    name: "akka-sse"
-    uri:  ${vars.uris.akka-sse-uri}
-    extra.exclude: ["jmh"]  // no benchmarking stuff plz
   }
 
   // this was added because conductr-lib's Play integration depended on it,
@@ -879,6 +869,12 @@ build += {
   ${vars.base} {
     name: "akka-contrib-extra"
     uri:  ${vars.uris.akka-contrib-extra-uri}
+    extra.options: ["-Dbintray.user=dummy", "-Dbintray.pass=dummy"]
+    extra.commands: ${vars.default-commands} [
+      "set every bintrayReleaseOnPublish := false"
+    ]
+    // Failed tests: akka.contrib.http.DirectivesSpec
+    extra.test-tasks: "compile"
   }
 
   ${vars.base} {
@@ -1212,6 +1208,8 @@ build += {
   ${vars.base} {
     name: "gigahorse"
     uri:  ${vars.uris.gigahorse-uri}
+    // as of August 2017, doesn't compile against latest akka-http
+    extra.exclude: ["akkaHttp"]
   }
 
   ${vars.base} {

--- a/configs/project-refs.conf
+++ b/configs/project-refs.conf
@@ -4,11 +4,10 @@
 
 vars.uris: {
   acyclic-uri:                  "https://github.com/lihaoyi/acyclic.git"
-  akka-contrib-extra-uri:       "https://github.com/scalacommunitybuild/akka-contrib-extra.git#community-build-2.12"
-  akka-http-uri:                "https://github.com/scalacommunitybuild/akka-http.git#community-build-2.12"
-  akka-more-uri:                "https://github.com/akka/akka.git#release-2.4"
-  akka-sse-uri:                 "https://github.com/scalacommunitybuild/akka-sse.git#community-build-2.12"
-  akka-uri:                     "https://github.com/akka/akka.git#release-2.4"
+  akka-actor-uri:               "https://github.com/akka/akka.git"
+  akka-contrib-extra-uri:       "https://github.com/typesafehub/akka-contrib-extra.git"
+  akka-http-uri:                "https://github.com/akka/akka-http.git"
+  akka-more-uri:                "https://github.com/akka/akka.git"
   algebra-uri:                  "https://github.com/typelevel/algebra.git"
   ammonite-uri:                 "https://github.com/lihaoyi/ammonite.git"
   argonaut-uri:                 "https://github.com/argonaut-io/argonaut.git"

